<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>

<title>nfcore/rnaseq Output</title>

<script type="text/javascript">
window.onload = function() {
  var imgs = document.getElementsByTagName('img'), i, img;
  for (i = 0; i < imgs.length; i++) {
    img = imgs[i];
    // center an image if it is the only element of its parent
    if (img.parentElement.childElementCount === 1)
      img.parentElement.style.textAlign = 'center';
  }
};
</script>





<style type="text/css">
body, td {
   font-family: sans-serif;
   background-color: white;
   font-size: 13px;
}

body {
  max-width: 800px;
  margin: auto;
  padding: 1em;
  line-height: 20px;
}

tt, code, pre {
   font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}

h1 {
   font-size:2.2em;
}

h2 {
   font-size:1.8em;
}

h3 {
   font-size:1.4em;
}

h4 {
   font-size:1.0em;
}

h5 {
   font-size:0.9em;
}

h6 {
   font-size:0.8em;
}

a:visited {
   color: rgb(50%, 0%, 50%);
}

pre, img {
  max-width: 100%;
}
pre {
  overflow-x: auto;
}
pre code {
   display: block; padding: 0.5em;
}

code {
  font-size: 92%;
  border: 1px solid #ccc;
}

code[class] {
  background-color: #F8F8F8;
}

table, td, th {
  border: none;
}

blockquote {
   color:#666666;
   margin:0;
   padding-left: 1em;
   border-left: 0.5em #EEE solid;
}

hr {
   height: 0px;
   border-bottom: none;
   border-top-width: thin;
   border-top-style: dotted;
   border-top-color: #999999;
}

@media print {
   * {
      background: transparent !important;
      color: black !important;
      filter:none !important;
      -ms-filter: none !important;
   }

   body {
      font-size:12pt;
      max-width:100%;
   }

   a, a:visited {
      text-decoration: underline;
   }

   hr {
      visibility: hidden;
      page-break-before: always;
   }

   pre, blockquote {
      padding-right: 1em;
      page-break-inside: avoid;
   }

   tr, img {
      page-break-inside: avoid;
   }

   img {
      max-width: 100% !important;
   }

   @page :left {
      margin: 15mm 20mm 15mm 10mm;
   }

   @page :right {
      margin: 15mm 10mm 15mm 20mm;
   }

   p, h2, h3 {
      orphans: 3; widows: 3;
   }

   h2, h3 {
      page-break-after: avoid;
   }
}
 
body {
  padding: 3em;
  margin-right: 350px;
  max-width: 100%;
}
#toc {
  position: fixed;
  right: 20px;
  width: 300px;
  padding-top: 20px;
  overflow: scroll;
  height: calc(100% - 3em - 20px);
}
#toc_header {
  font-size: 1.8em;
  font-weight: bold;
}
#toc > ul {
  padding-left: 0;
  list-style-type: none;
}
#toc > ul ul { padding-left: 20px; }
#toc > ul > li > a { display: none; }
img { max-width: 800px; }

</style>



</head>

<body>
<div id="toc">
<div id="toc_header">Table of Contents</div>
<ul>
<li>
<a href="#toc_0">nfcore/rnaseq Output</a>
<ul>
<li>
<a href="#toc_1">Pipeline overview</a>
</li>
<li>
<a href="#toc_2">FastQC</a>
</li>
<li>
<a href="#toc_3">TrimGalore</a>
</li>
<li>
<a href="#toc_4">STAR</a>
</li>
<li>
<a href="#toc_5">RSeQC</a>
<ul>
<li>
<a href="#toc_6">BAM stat</a>
</li>
<li>
<a href="#toc_7">Infer experiment</a>
</li>
<li>
<a href="#toc_8">Junction saturation</a>
</li>
<li>
<a href="#toc_9">RPKM saturation</a>
</li>
<li>
<a href="#toc_10">Read duplication</a>
</li>
<li>
<a href="#toc_11">Inner distance</a>
</li>
<li>
<a href="#toc_12">Gene body coverage</a>
</li>
<li>
<a href="#toc_13">Read distribution</a>
</li>
<li>
<a href="#toc_14">Junction annotation</a>
</li>
</ul>
</li>
<li>
<a href="#toc_15">dupRadar</a>
</li>
<li>
<a href="#toc_16">Preseq</a>
</li>
<li>
<a href="#toc_17">featureCounts</a>
</li>
<li>
<a href="#toc_18">StringTie</a>
</li>
<li>
<a href="#toc_19">Sample Correlation</a>
</li>
<li>
<a href="#toc_20">MultiQC</a>
</li>
</ul>
</li>
</ul>
</div>


<h1 id="toc_0">nfcore/rnaseq Output</h1>

<p>nfcore/rnaseq is an RNA-seq analysis pipeline. This document describes the output produced by the pipeline.</p>

<p>Most of the plots are taken from the MultiQC report, which summarises results at the end of the pipeline.</p>

<h2 id="toc_1">Pipeline overview</h2>

<p>The pipeline is built using <a href="https://www.nextflow.io/">Nextflow</a>
and processes data using the following steps:</p>

<ul>
<li><a href="#fastqc">FastQC</a> - read quality control</li>
<li><a href="#trimgalore">TrimGalore</a> - adapter trimming</li>
<li><a href="#star">STAR</a> - alignment</li>
<li><a href="#rseqc">RSeQC</a> - RNA quality control metrics

<ul>
<li><a href="#bam-stat">BAM stat</a></li>
<li><a href="#infer-experiment">Infer experiment</a></li>
<li><a href="#junction-saturation">Junction saturation</a></li>
<li><a href="#rpkm-saturation">RPKM saturation</a></li>
<li><a href="#read-duplication">Read duplication</a></li>
<li><a href="#inner-distance">Inner distance</a></li>
<li><a href="#gene-body-coverage">Gene body coverage</a></li>
<li><a href="#read-distribution">Read distribution</a></li>
<li><a href="#junction-annotation">Junction annotation</a></li>
</ul></li>
<li><a href="#dupradar">dupRadar</a> - technical / biological read duplication</li>
<li><a href="#preseq">Preseq</a> - library complexity</li>
<li><a href="#featurecounts">featureCounts</a> - gene counts, biotype counts, rRNA estimation.</li>
<li><a href="#stringtie">StringTie</a> - FPKMs for genes and transcripts</li>
<li><a href="#Sample_correlation">Sample_correlation</a> - create MDS plot and sample pairwise distance heatmap / dendrogram</li>
<li><a href="#multiqc">MultiQC</a> - aggregate report, describing results of the whole pipeline</li>
</ul>

<h2 id="toc_2">FastQC</h2>

<p><a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> gives general quality metrics about your reads. It provides information about the quality score distribution across your reads, the per base sequence content (%T/A/G/C). You get information about adapter contamination and other overrepresented sequences.</p>

<p>For further reading and documentation see the <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/">FastQC help</a>.</p>

<blockquote>
<p><strong>NB:</strong> The FastQC plots displayed in the MultiQC report shows <em>untrimmed</em> reads. They may contain adapter sequence and potentially regions with low quality. To see how your reads look after trimming, look at the FastQC reports in the <code>trim_galore</code> directory.</p>
</blockquote>

<p><strong>Output directory: <code>results/fastqc</code></strong></p>

<ul>
<li><code>sample_fastqc.html</code>

<ul>
<li>FastQC report, containing quality metrics for your untrimmed raw fastq files</li>
</ul></li>
<li><code>zips/sample_fastqc.zip</code>

<ul>
<li>zip file containing the FastQC report, tab-delimited data file and plot images</li>
</ul></li>
</ul>

<h2 id="toc_3">TrimGalore</h2>

<p>The nfcore/rnaseq pipeline uses <a href="http://www.bioinformatics.babraham.ac.uk/projects/trim_galore/">TrimGalore</a> for removal of adapter contamination and trimming of low quality regions. TrimGalore uses <a href="https://github.com/marcelm/cutadapt">Cutadapt</a> for adapter trimming and runs FastQC after it finishes.</p>

<p>MultiQC reports the percentage of bases removed by TrimGalore in the <em>General Statistics</em> table, along with a line plot showing where reads were trimmed.</p>

<p><strong>Output directory: <code>results/trim_galore</code></strong></p>

<p>Contains FastQ files with quality and adapter trimmed reads for each sample, along with a log file describing the trimming.</p>

<ul>
<li><code>sample_val_1.fq.gz</code>, <code>sample_val_2.fq.gz</code>

<ul>
<li>Trimmed FastQ data, reads 1 and 2.</li>
<li>NB: Only saved if <code>--saveTrimmed</code> has been specified.</li>
</ul></li>
<li><code>logs/sample_val_1.fq.gz_trimming_report.txt</code>

<ul>
<li>Trimming report (describes which parameters that were used)</li>
</ul></li>
<li><code>FastQC/sample_val_1_fastqc.zip</code>

<ul>
<li>FastQC report for trimmed reads</li>
</ul></li>
</ul>

<p>Single-end data will have slightly different file names and only one FastQ file per sample.</p>

<h2 id="toc_4">STAR</h2>

<p>STAR is a read aligner designed for RNA sequencing.  STAR stands for Spliced Transcripts Alignment to a Reference, it produces results comparable to TopHat (the aligned previously used by NGI for RNA alignments) but is much faster.</p>

<p>The STAR section of the MultiQC report shows a bar plot with alignment rates: good samples should have most reads as <em>Uniquely mapped</em> and few <em>Unmapped</em> reads.</p>

<p><img src="images/star_alignment_plot.png" alt="STAR"></p>

<p><strong>Output directory: <code>results/STAR</code></strong></p>

<ul>
<li><code>Sample_Aligned.sortedByCoord.out.bam</code>

<ul>
<li>The aligned BAM file</li>
</ul></li>
<li><code>Sample_Log.final.out</code>

<ul>
<li>The STAR alignment report, contains mapping results summary</li>
</ul></li>
<li><code>Sample_Log.out</code> and <code>Sample_Log.progress.out</code>

<ul>
<li>STAR log files, containing a lot of detailed information about the run. Typically only useful for debugging purposes.</li>
</ul></li>
<li><code>Sample_SJ.out.tab</code>

<ul>
<li>Filtered splice junctions detected in the mapping</li>
</ul></li>
</ul>

<h2 id="toc_5">RSeQC</h2>

<p>RSeQC is a package of scripts designed to evaluate the quality of RNA seq data. You can find out more about the package at the <a href="http://rseqc.sourceforge.net/">RSeQC website</a>.</p>

<p>This pipeline runs several, but not all RSeQC scripts. All of these results are summarised within the MultiQC report and described below.</p>

<p><strong>Output directory: <code>results/rseqc</code></strong></p>

<p>These are all quality metrics files and contains the raw data used for the plots in the MultiQC report. In general, the <code>.r</code> files are R scripts for generating the figures, the <code>.txt</code> are summary files, the <code>.xls</code> are data tables and the <code>.pdf</code> files are summary figures.</p>

<h3 id="toc_6">BAM stat</h3>

<p><strong>Output: <code>Sample_bam_stat.txt</code></strong></p>

<p>This script gives numerous statistics about the aligned BAM files produced by STAR. A typical output looks as follows:</p>

<pre><code>#Output (all numbers are read count)
#==================================================
Total records:                                 41465027
QC failed:                                     0
Optical/PCR duplicate:                         0
Non Primary Hits                               8720455
Unmapped reads:                                0

mapq &lt; mapq_cut (non-unique):                  3127757
mapq &gt;= mapq_cut (unique):                     29616815
Read-1:                                        14841738
Read-2:                                        14775077
Reads map to &#39;+&#39;:                              14805391
Reads map to &#39;-&#39;:                              14811424
Non-splice reads:                              25455360
Splice reads:                                  4161455
Reads mapped in proper pairs:                  21856264
Proper-paired reads map to different chrom:    7648
</code></pre>

<p>MultiQC plots each of these statistics in a dot plot. Each sample in the project is a dot - hover to see the sample highlighted across all fields.</p>

<p>RSeQC documentation: <a href="http://rseqc.sourceforge.net/#bam-stat-py">bam_stat.py</a></p>

<h3 id="toc_7">Infer experiment</h3>

<p><strong>Output: <code>Sample_infer_experiment.txt</code></strong></p>

<p>This script predicts the mode of library preparation (sense-stranded or antisense-stranded) according to how aligned reads overlay gene features in the reference genome.
Example output from an unstranded (~50% sense/antisense) library of paired end data:</p>

<p><strong>From MultiQC report:</strong>
<img src="images/rseqc_infer_experiment_plot.png" alt="infer_experiment"></p>

<p><strong>From the <code>infer_experiment.txt</code> file:</strong></p>

<pre><code>This is PairEnd Data
Fraction of reads failed to determine: 0.0409
Fraction of reads explained by &quot;1++,1--,2+-,2-+&quot;: 0.4839
Fraction of reads explained by &quot;1+-,1-+,2++,2--&quot;: 0.4752
</code></pre>

<p>RSeQC documentation: <a href="http://rseqc.sourceforge.net/#infer-experiment-py">infer_experiment.py</a></p>

<h3 id="toc_8">Junction saturation</h3>

<p><strong>Output:</strong></p>

<ul>
<li><code>Sample_rseqc.junctionSaturation_plot.pdf</code></li>
<li><code>Sample_rseqc.junctionSaturation_plot.r</code></li>
</ul>

<p>This script shows the number of splice sites detected at the data at various levels of subsampling. A sample that reaches a plateau before getting to 100% data indicates that all junctions in the library have been detected, and that further sequencing will not yield more observations. A good sample should approach such a plateau of <em>Known junctions</em>, very deep sequencing is typically requires to saturate all <em>Novel Junctions</em> in a sample.</p>

<p>None of the lines in this example have plateaued and thus these samples could reveal more alternative splicing information if they were sequenced deeper.</p>

<p><img src="images/rseqc_junction_saturation_plot.png" alt="Junction saturation"></p>

<p>RSeQC documentation: <a href="http://rseqc.sourceforge.net/#junction-saturation-py">junction_saturation.py</a></p>

<h3 id="toc_9">RPKM saturation</h3>

<p><strong>Output:</strong></p>

<ul>
<li><code>Sample_RPKM_saturation.eRPKM.xls</code></li>
<li><code>Sample_RPKM_saturation.rawCount.xls</code></li>
<li><code>Sample_RPKM_saturation.saturation.pdf</code></li>
<li><code>Sample_RPKM_saturation.saturation.r</code></li>
</ul>

<p>This tool resamples a subset of the total RNA reads and calculates the RPKM value for each subset. We use the default subsets of every 5% of the total reads.
A percent relative error is then calculated based on the subsamples; this is the y-axis in the graph. A typical PDF figure looks as follows:</p>

<p><img src="images/saturation.png" alt="RPKM saturation"></p>

<p>A complex library will have low resampling error in well expressed genes.</p>

<p>This data is not currently reported in the MultiQC report.</p>

<p>RSeQC documentation: <a href="http://rseqc.sourceforge.net/#rpkm-saturation-py">RPKM_saturation.py</a></p>

<h3 id="toc_10">Read duplication</h3>

<p><strong>Output:</strong></p>

<ul>
<li><code>Sample_read_duplication.DupRate_plot.pdf</code></li>
<li><code>Sample_read_duplication.DupRate_plot.r</code></li>
<li><code>Sample_read_duplication.pos.DupRate.xls</code></li>
<li><code>Sample_read_duplication.seq.DupRate.xls</code></li>
</ul>

<p>This plot shows the number of reads (y-axis) with a given number of exact duplicates (x-axis). Most reads in an RNA-seq library should have a low number of exact duplicates. Samples which have many reads with many duplicates (a large area under the curve) may be suffering excessive technical duplication.</p>

<p><img src="images/rseqc_read_dups_plot.png" alt="Read duplication"></p>

<p>RSeQC documentation: <a href="http://rseqc.sourceforge.net/#read-duplication-py">read_duplication.py</a></p>

<h3 id="toc_11">Inner distance</h3>

<p><strong>Output:</strong></p>

<ul>
<li><code>Sample_rseqc.inner_distance.txt</code></li>
<li><code>Sample_rseqc.inner_distance_freq.txt</code></li>
<li><code>Sample_rseqc.inner_distance_plot.r</code></li>
</ul>

<p>The inner distance script tries to calculate the inner distance between two paired RNA reads. It is the distance between the end of read 1 to the start of read 2,
and it is sometimes confused with the insert size (see <a href="http://thegenomefactory.blogspot.com.au/2013/08/paired-end-read-confusion-library.html">this blog post</a> for disambiguation):
<img src="images/inner_distance_concept.png" alt="inner distance concept"></p>

<blockquote>
<p><em>Credit: modified from RSeQC documentation.</em></p>
</blockquote>

<p>Note that values can be negative if the reads overlap. A typical set of samples may look like this:
<img src="images/rseqc_inner_distance_plot.png" alt="Inner distance"></p>

<p>This plot will not be generated for single-end data. Very short inner distances are often seen in old or degraded samples (<em>eg.</em> FFPE).</p>

<p>RSeQC documentation: <a href="http://rseqc.sourceforge.net/#inner-distance-py">inner_distance.py</a></p>

<h3 id="toc_12">Gene body coverage</h3>

<p><strong>NB:</strong> In nfcore/rnaseq we subsample this to 1 Million reads. This speeds up this task significantly and has no to little effect on the results.</p>

<p><strong>Output:</strong></p>

<ul>
<li><code>Sample_rseqc.geneBodyCoverage.curves.pdf</code></li>
<li><code>Sample_rseqc.geneBodyCoverage.r</code></li>
<li><code>Sample_rseqc.geneBodyCoverage.txt</code></li>
</ul>

<p>This script calculates the reads coverage across gene bodies. This makes it easy to identify 3&#39; or 5&#39; skew in libraries. A skew towards increased 3&#39; coverage can happen in degraded samples prepared with poly-A selection.</p>

<p>A typical set of libraries with little or no bias will look as follows:</p>

<p><img src="images/rseqc_gene_body_coverage_plot.png" alt="Gene body coverage"></p>

<p>RSeQC documentation: <a href="http://rseqc.sourceforge.net/#genebody-coverage-py">gene_body_coverage.py</a></p>

<h3 id="toc_13">Read distribution</h3>

<p><strong>Output: <code>Sample_read_distribution.txt</code></strong></p>

<p>This tool calculates how mapped reads are distributed over genomic features. A good result for a standard RNA seq experiments is generally to have as many exonic reads as possible (<code>CDS_Exons</code>). A large amount of intronic reads could be indicative of DNA contamination in your sample or some other problem.</p>

<p><img src="images/rseqc_read_distribution_plot.png" alt="Read distribution"></p>

<p>RSeQC documentation: <a href="http://rseqc.sourceforge.net/#read-distribution-py">read_distribution.py</a></p>

<h3 id="toc_14">Junction annotation</h3>

<p><strong>Output:</strong></p>

<ul>
<li><code>Sample_junction_annotation_log.txt</code></li>
<li><code>Sample_rseqc.junction.xls</code></li>
<li><code>Sample_rseqc.junction_plot.r</code></li>
<li><code>Sample_rseqc.splice_events.pdf</code></li>
<li><code>Sample_rseqc.splice_junction.pdf</code></li>
</ul>

<p>Junction annotation compares detected splice junctions to a reference gene model. An RNA read can be spliced 2 or more times, each time is called a splicing event.</p>

<p><img src="images/rseqc_junction_annotation_junctions_plot.png" alt="Junction annotation"></p>

<p>RSeQC documentation: <a href="http://rseqc.sourceforge.net/#junction-annotation-py">junction_annotation.py</a></p>

<h2 id="toc_15">dupRadar</h2>

<p><a href="https://www.bioconductor.org/packages/release/bioc/html/dupRadar.html">dupRadar</a> is a Bioconductor library for R. It plots the duplication rate against expression (RPKM) for every gene. A good sample with little technical duplication will only show high numbers of duplicates for highly expressed genes. Samples with technical duplication will have high duplication for all genes, irrespective of transcription level.</p>

<p><img src="images/dupRadar_plot.png" alt="dupRadar"></p>

<blockquote>
<p><em>Credit: <a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/dupRadar/inst/doc/dupRadar.html">dupRadar documentation</a></em></p>
</blockquote>

<p><strong>Output directory: <code>results/dupRadar</code></strong></p>

<ul>
<li><code>Sample_markDups.bam_duprateExpDens.pdf</code></li>
<li><code>Sample_markDups.bam_duprateExpBoxplot.pdf</code></li>
<li><code>Sample_markDups.bam_expressionHist.pdf</code></li>
<li><code>Sample_markDups.bam_dupMatrix.txt</code></li>
<li><code>Sample_markDups.bam_duprateExpDensCurve.txt</code></li>
<li><code>Sample_markDups.bam_intercept_slope.txt</code></li>
</ul>

<p>DupRadar documentation: <a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/dupRadar/inst/doc/dupRadar.html">dupRadar docs</a></p>

<h2 id="toc_16">Preseq</h2>

<p><a href="http://smithlabresearch.org/software/preseq/">Preseq</a> estimates the complexity of a library, showing how many additional unique reads are sequenced for increasing the total read count. A shallow curve indicates that the library has reached complexity saturation and further sequencing would likely not add further unique reads. The dashed line shows a perfectly complex library where total reads = unique reads.</p>

<p>Note that these are predictive numbers only, not absolute. The MultiQC plot can sometimes give extreme sequencing depth on the X axis - click and drag from the left side of the plot to zoom in on more realistic numbers.</p>

<p><img src="images/preseq_plot.png" alt="preseq"></p>

<p><strong>Output directory: <code>results/preseq</code></strong></p>

<ul>
<li><code>sample_ccurve.txt</code>

<ul>
<li>This file contains plot values for the complexity curve, plotted in the MultiQC report.</li>
</ul></li>
</ul>

<h2 id="toc_17">featureCounts</h2>

<p><a href="http://bioinf.wehi.edu.au/featureCounts/">featureCounts</a> from the subread package summarises the read distribution over genomic features such as genes, exons, promotors, gene bodies, genomic bins and chromosomal locations.
RNA reads should mostly overlap genes, so be assigned.</p>

<p><img src="images/featureCounts_assignment_plot.png" alt="featureCounts"></p>

<p>We also use featureCounts to count overlaps with different classes of features. This gives a good idea of where aligned reads are ending up and can show potential problems such as rRNA contamination.
<img src="images/featureCounts_biotype_plot.png" alt="biotypes"></p>

<p><strong>Output directory: <code>results/featureCounts</code></strong></p>

<ul>
<li><code>Sample.bam_biotype_counts.txt</code>

<ul>
<li>Read counts for the different gene biotypes that featureCounts distinguishes.</li>
</ul></li>
<li><code>Sample.featureCounts.txt</code>

<ul>
<li>Read the counts for each gene provided in the reference <code>gtf</code> file</li>
</ul></li>
<li><code>Sample.featureCounts.txt.summary</code>

<ul>
<li>Summary file, containing statistics about the counts</li>
</ul></li>
</ul>

<h2 id="toc_18">StringTie</h2>

<p><a href="https://ccb.jhu.edu/software/stringtie/">StringTie</a> assembles RNA-Seq alignments into potential transcripts. It assembles and quantitates full-length transcripts representing multiple splice variants for each gene locus.</p>

<p>StringTie outputs FPKM metrics for genes and transcripts as well as the transcript features that it generates.</p>

<p><strong>Output directory: <code>results/stringtie</code></strong></p>

<ul>
<li><code>&lt;sample&gt;_Aligned.sortedByCoord.out.bam.gene_abund.txt</code>

<ul>
<li>Gene aboundances, FPKM values</li>
</ul></li>
<li><code>&lt;sample&gt;_Aligned.sortedByCoord.out.bam_transcripts.gtf</code>

<ul>
<li>This <code>.gtf</code> file contains all of the assembled transcipts from StringTie</li>
</ul></li>
<li><code>&lt;sample&gt;_Aligned.sortedByCoord.out.bam.cov_refs.gtf</code>

<ul>
<li>This <code>.gtf</code> file contains the transcripts that are fully covered by reads.</li>
</ul></li>
</ul>

<h2 id="toc_19">Sample Correlation</h2>

<p><a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html">edgeR</a> is a Bioconductor package for R used for RNA-seq data analysis. The script included in the pipeline uses edgeR to normalise read counts and create a heatmap / dendrogram showing pairwise euclidean distance (sample similarity). It also creates a 2D MDS scatter plot showing sample grouping. These help to show sample similarity and can reveal batch effects and sample groupings.</p>

<p><strong>Heatmap:</strong></p>

<p><img src="images/mqc_hcplot_hocmzpdjsq.png" alt="heatmap"></p>

<p><strong>MDS plot:</strong></p>

<p><img src="images/mqc_hcplot_ltqchiyxfz.png" alt="mds_plot"></p>

<p><strong>Output directory: <code>results/sample_correlation</code></strong></p>

<ul>
<li><code>edgeR_MDS_plot.pdf</code>

<ul>
<li>MDS scatter plot, showing sample similarity</li>
</ul></li>
<li><code>edgeR_MDS_distance_matrix.txt</code>

<ul>
<li>Distance matrix containing raw data from MDS analysis</li>
</ul></li>
<li><code>edgeR_MDS_plot_coordinates.txt</code>

<ul>
<li>Scatter plot coordinates from MDS plot, used for MultiQC report</li>
</ul></li>
<li><code>log2CPM_sample_distances_dendrogram.pdf</code>

<ul>
<li>Dendrogram plot showing the euclidian distance between your samples</li>
</ul></li>
<li><code>log2CPM_sample_distances_heatmap.pdf</code>

<ul>
<li>Heatmap plot showing the euclidian distance between your samples</li>
</ul></li>
<li><code>log2CPM_sample_distances.txt</code>

<ul>
<li>Raw data used for heatmap and dendrogram plots.</li>
</ul></li>
</ul>

<h2 id="toc_20">MultiQC</h2>

<p><a href="http://multiqc.info">MultiQC</a> is a visualisation tool that generates a single HTML report summarising all samples in your project. Most of the pipeline QC results are visualised in the report and further statistics are available in within the report data directory.</p>

<p><strong>Output directory: <code>results/multiqc</code></strong></p>

<ul>
<li><code>Project_multiqc_report.html</code>

<ul>
<li>MultiQC report - a standalone HTML file that can be viewed in your web browser</li>
</ul></li>
<li><code>Project_multiqc_data/</code>

<ul>
<li>Directory containing parsed statistics from the different tools used in the pipeline</li>
</ul></li>
</ul>

<p>For more information about how to use MultiQC reports, see <a href="http://multiqc.info">http://multiqc.info</a></p>

</body>

</html>
